kong/plugins/grpc-gateway/deco.lua

```diff
@@ -220,11 +249,39 @@ function deco:upstream(body)
     // which don't use the URL at all for transferring data.
   ]]
   -- TODO: do we allow http parameter when body is not *?
-  local payload = self.template_payload
+  --local payload = self.template_payload
+  local payload = setmetatable( self.template_payload, {
+  --[[
+    // Set value at dot-separated path. For example: `GET /v1/messages/123456?revision=2&sub.subfield=foo`
+    // translates into `payload = { sub = { subfield = "foo" }}`
+    //
+    // This is to comply with [spec](https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L113),
+    // where also fields of messages can be passed in query string.
+    //
+  ]]--
+  __newindex = function (t, k, v)
```

Comment:
Consider moving this function and the metatable out of the function body. Instead of creating the metatable every time, we should reuse them.

Reply:
https://github.com/Kong/kong/pull/8297/commits/d5c17bd12cd99db44baef7e59a8871d3013d81e0